tensornetwork/tensor.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+import math
+import copy
+import warnings
+from typing import Any, Union, Text, Optional, List, Sequence
+import numpy as np
+from tensornetwork.backends import base_backend
+from tensornetwork import backends, backend_contextmanager
+
+BaseBackend = base_backend.BaseBackend
+
+class Tensor():
+  def __init__(self,
+               array: Any,
+               backend: Optional[Union[Text, BaseBackend]] = None) -> None:
+    if backend is None:
+      backend = backend_contextmanager.get_default_backend()
+    backend_obj = backends.backend_factory.get_backend(backend)
+    self.backend = backend_obj
+    self.array = self.backend.convert_to_tensor(array)
+    self.shape = array.shape
+    self.size = np.prod(self.shape)
+    self.ndim = len(self.shape)
+
+  #def __getitem__
+  #def __setitem__
+
+  @property
+  def dtype(self) -> Any: # To maintain backend independence
+    """ Returns: The dtype of the backend array.
+    """
+    return self.array.dtype
+
+  @property
+  def T(self) -> Tensor:
+    """ The `Tensor` with reversed axes.
+    Returns:
+      The transposed `Tensor`.
+    """
+    return self.transpose()
+
+  @property
+  def H(self) -> Tensor:
+    """ The conjugate `Tensor` with reversed axes.
+    """
+    star = self.backend.conj(self.array)
+    array_H = self.backend.transpose(star)
+    return Tensor(array_H, backend=self.backend)
+
+  #  @property
+  #  def real(self) -> Tensor:
+  #    """ Returns: The real part of the `Tensor`.
+  #    """
+  #    array_r = self.backend.real(self.array)
+  #    return Tensor(array_r, backend=self.backend)
+
+  #  @property
+  #  def imag(self) -> Tensor:
+  #    """ Returns: The imaginary part of the `Tensor`.
+  #    """
+  #    array_i = self.backend.imag(self.array)
+  #    return Tensor(array_i, backend=self.backend)
+
+  # def all(self):
+
+  # def any(self):
+
+  # def argmax():
+
+  # def argmin():
```

Comment:
In general, always delete commented out code.

If you're adding these now to be made later, mark them with `TODO(alewis): Implement the following methods`.